manifest/etcd.yaml

```diff
@@ -0,0 +1,148 @@
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: etcd
+spec:
+  secretName: etcd-secret
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  subject:
+    organizations:
+      - redhat
+  isCA: false
+  privateKey:
+    algorithm: RSA
+    encoding: PKCS1
+    size: 2048
+  usages:
+    - server auth
+    - client auth
+  dnsNames:
+    - etcd
+    - etcd-0
+    - etcd-1
+    - etcd-2
+    - etcd-0.etcd
+    - etcd-1.etcd
+    - etcd-2.etcd
+  ipAddresses:
+    - 0.0.0.0
+  issuerRef:
+    name: kcp
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: etcd-peer
+spec:
+  secretName: etcd-peer-secret
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
+  subject:
+    organizations:
+      - redhat
+  isCA: false
+  privateKey:
+    algorithm: RSA
+    encoding: PKCS1
+    size: 2048
+  usages:
+    - server auth
+    - client auth
+  dnsNames:
+    - etcd-0.etcd
+    - etcd-1.etcd
+    - etcd-2.etcd
+    - etcd-0
+    - etcd-1
+    - etcd-2
+  ipAddresses:
+    - 0.0.0.0
+  issuerRef:
+    name: kcp
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: etcd
+spec:
+  clusterIP: None
+  ports:
+    - port: 2379
+      name: client
+    - port: 2380
+      name: peer
+  selector:
+    app: etcd
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: etcd
+  labels:
+    app: etcd
+spec:
+  serviceName: etcd
+  selector:
+    matchLabels:
+      app: etcd
+  replicas: 3
+  template:
+    metadata:
+      name: etcd
+      labels:
+        app: etcd
+    spec:
+      containers:
+        - name: etcd
+          image: quay.io/coreos/etcd:v3.5.2
+          ports:
+            - containerPort: 2379
+              name: client
+            - containerPort: 2380
+              name: peer
+          volumeMounts:
+            - name: etcd-data
+              mountPath: /var/run/etcd
+            - name: peer-certs
+              mountPath: /etc/etcd/tls/peer
+            - name: server-certs
+              mountPath: /etc/etcd/tls/server
+          command:
+            - /bin/sh
+            - -c
+            - |
+              PEERS="etcd-0=https://etcd-0.etcd:2380,etcd-1=https://etcd-1.etcd:2380,etcd-2=https://etcd-2.etcd:2380"
+              exec etcd --name ${HOSTNAME} \
+                --listen-peer-urls https://0.0.0.0:2380 \
+                --initial-advertise-peer-urls https://${HOSTNAME}:2380 \
+                --listen-client-urls https://0.0.0.0:2379 \
```

Comment:
nit: can we restrict this somehow? `0.0.0.0` seems very open

Reply:
We tested setting all the hostnames to `${HOSTNAME}`, and it caused some binding conflicts.  Binding to `127.0.0.1` didn't expose the port to KCP.  FWIW the etcd running in our pcluster uses 0.0.0.0.